x-pack/plugins/cases/public/components/all_cases/use_url_state.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+import { isEqual } from 'lodash';
+
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { stringify } from 'query-string';
+
+import { DEFAULT_QUERY_PARAMS } from '../../containers/use_get_cases';
+import { parseUrlQueryParams } from './utils';
+
+import type {
+  LocalStorageQueryParams,
+  QueryParams,
+  UrlQueryParams,
+} from '../../../common/ui/types';
+
+export function useUrlState(isModalView: boolean = false) {
+  const location = useLocation();
+  const history = useHistory();
+
+  const [queryParams, setQueryParams] = useState<UrlQueryParams>({});
+
+  const [localStorageQueryParams, setLocalStorageQueryParams] =
+    useLocalStorage<LocalStorageQueryParams>('cases.list.preferences');
```

Comment:
Cases are rendered in Observabliilty, Security solution, and the Stack. IMO, we should not have one setting for all solutions. What about we namespace the local storage key with the `appId`? Example:

```
const { appId } = useCasesContext();

const [localStorageQueryParams, setLocalStorageQueryParams] =
    useLocalStorage<LocalStorageQueryParams>(getQueryParamsLocalStorageKey(appId));
```

where `getQueryParamsLocalStorageKey`

```
const getQueryParamsLocalStorageKey = (appId) => {
   const filteringKey = LOCAL_STORAGE_KEYS['casesFiltering']
   return `${appId}.${filteringKey}`
}
```

The `LOCAL_STORAGE_KEYS` should be a map/record that contains all local storage keys. It should be placed in `x-pack/plugins/cases/common/constants.ts`. Example:

```
const LOCAL_STORAGE_KEYS = { 'casesFiltering': 'cases.list.filtering' } 
```